jetty-io/src/main/java/org/eclipse/jetty/io/ConnectionStatistics.java

```diff
@@ -78,159 +65,312 @@ public void onOpened(Connection connection)
     {
         if (!isStarted())
             return;
-
-        _connections.increment();
+        _stats.incrementCount();
+        _statsMap.computeIfAbsent(connection.getClass().getName(), Stats::new).incrementCount();
```

Comment:
perhaps delegate the computeIfAbsent to a protected method, so that if somebody really wanted to pick and choose which types get counted then they could override that method and only return a Stats if they wanted that specific connection counted and it might be on a different criteria than className.